src/main/java/dk/cngroup/lentils/logger/Logger.java

```diff
@@ -0,0 +1,188 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+
+import static dk.cngroup.lentils.entity.CypherStatus.SKIPPED;
+import static dk.cngroup.lentils.logger.utils.LoggerUtils.getChangeCypherStatusPoints;
+import static dk.cngroup.lentils.logger.utils.LoggerUtils.getRevertHintPoints;
+import static dk.cngroup.lentils.logger.utils.LoggerUtils.getTakeHintPoints;
+import static dk.cngroup.lentils.logger.utils.LoggerUtils.getVerifyCodewordPoints;
+
+@Aspect
+@Component
+public class Logger {
+
+    private final Printer printer;
+    private final ScoreService scoreService;
+    private final HintService hintService;
+    private final TeamService teamService;
+    private final CypherService cypherService;
+    private final StatusService statusService;
+    private final HintTakenService hintTakenService;
+
+    @Autowired
+    public Logger(final Printer printer,
+                  final ScoreService scoreService,
+                  final HintService hintService,
+                  final TeamService teamService,
+                  final CypherService cypherService,
+                  final StatusService statusService,
+                  final HintTakenService hintTakenService) {
+        this.printer = printer;
+        this.scoreService = scoreService;
+        this.hintService = hintService;
+        this.teamService = teamService;
+        this.cypherService = cypherService;
+        this.statusService = statusService;
+        this.hintTakenService = hintTakenService;
+    }
+
+    /**
+     * CLIENT METHODS.
+     */
+    @Around("execution(* dk.cngroup.lentils.controller.ClientController.verifyCodeword(..))" +
+            "&& args(cypherId,codeword,user,..)")
+    public Object verifyCodeword(final ProceedingJoinPoint joinPoint,
+                                 final Long cypherId,
+                                 final Codeword codeword,
+                                 final CustomUserDetails user) throws Throwable {
+        String result = (String) joinPoint.proceed(joinPoint.getArgs());
+
+        int points = getVerifyCodewordPoints(result);
+        int score = scoreService.getScoreByTeam(user.getTeam());
+
+        Message<Codeword> message = MessageFactory.createVerifyCodeword(user, codeword, points, score);
+        printer.println(message);
+
+        return result;
+    }
+
+    @After("execution(* dk.cngroup.lentils.controller.ClientController.getHint(..))" +
+            "&& args(hintId,user,..)")
+    public void getHint(final Long hintId,
+                        final CustomUserDetails user) {
+
+        Hint hint = hintService.getHint(hintId);
+        int points = -hint.getValue();
+        int score = scoreService.getScoreByTeam(user.getTeam());
+
+        Message<Long> message = MessageFactory.createTakeHint(user, hintId, points, score);
```

Comment:
otazka :)
je lepsi si nejdriv ulozit data do promenne a ty pak pouzit pri volani metody tak jak to mas ty:
```
        Hint hint = hintService.getHint(hintId);
        int points = -hint.getValue();
        int score = scoreService.getScoreByTeam(user.getTeam());

        Message<Long> message = MessageFactory.createTakeHint(user, hintId, points, score);
```
nebo je lepsi je volat primo v metode, napriklad:

`Message<Long> message = MessageFactory.createTakeHint(user, hintId, -hint.getValue(), scoreService.getScoreByTeam(user.getTeam()));`

Reply:
Přijde na to.
Pokud je to krátké, jsem pro ty pomocné proměnné nemít. Ale, pokud třeba getter moc nevypovídá o tom, co ten parametr vlastně znamená, je lepší mít pomocnou proměnnou s rozumným názvem a předat až tu.

Reply:
Hlavni je v dnesni dobe citelnost, takze si rekni sam, co se ti lip cte (mysleno tak, ze co snadnej a rychlej prectes a pochopis). Toto:
```java
Message<Long> message = MessageFactory.createTakeHint(user, hintId, points, score);
```
a nebo toto?
```java
Message<Long> message = MessageFactory.createTakeHint(user, hintId, -hint.getValue(), scoreService.getScoreByTeam(user.getTeam()));
```